eclipse/paho.mqtt.android
# Resolve confusing dependency management and Move from minSDK 9 to min 14

It seems this project was converted to gradle a while ago, and it was using some old "hacks" and work arounds. Especially the "double dependency" necessity for people and the use of "@aar". This is no longer necessary, and when you release a new version, clients (like the sample app) will only need to do:

`compile "org.eclipse.paho:org.eclipse.paho.android.service:X.X.X"`

What's changed:

- Upgrade to latest gradle build tools (3.0.0)
 - Remove unused code, and debug cradle code
 - Compile against latest Android (26)
 - Using latest build tools (26.0.0)
 - Use latest support library (26.1.0)
 - Add google maven repo
 - Use latest gradle version (4.1)
 - Target latest version of Android in samples (26)
 - Remove dependency on libs folder - unnecessary
 - Remove unused empty proguard files
 - Remove release flavours for samples (never used)
 - Moved the service and non-dangerous permissions to the library itself, now they will be included by default and less work for clients to do. 

The example and sample have to depend on the compile project dependency for now, until a new release is done that includes the manifest merging.

Next steps:

 - release a new version to update example and sample
 - look at the release process and consider using a library instead of the all the manual stuff you have [like Bintray release plugin](github.com/novoda/bintray-release)
 - look at the use of support-v4 and move to using a smaller modular dependency

Please make sure that the following boxes are checked before submitting your Pull Request, thank you!

- [X] You have signed the [Eclipse ECA](https://wiki.eclipse.org/ECA)
- [X] All of your commits have been signed-off with the correct email address (The same one that you used to sign the CLA)
- [X] If This PR fixes an issue, that you reference the issue below. OR if this is a new issue that you are fixing straight away that you add some Description about the bug and how this will fix it.
- [X] If this is new functionality, You have added the appropriate Unit tests.
